packages/client/src/rpc/validation.ts

```diff
@@ -425,6 +428,110 @@ export const validators = {
     }
   },
 
+  get depositRequest() {
+    return (
+      requiredFields: string[] = ['pubkey', 'withdrawalCredentials', 'amount', 'signature', 'index']
+    ) => {
+      return (params: any[], index: number) => {
+        if (typeof params[index] !== 'object') {
+          return {
+            code: INVALID_PARAMS,
+            message: `invalid argument ${index}: argument must be an object`,
+          }
+        }
+
+        const wt = params[index]
+
+        for (const field of requiredFields) {
+          if (wt[field] === undefined) {
+            return {
+              code: INVALID_PARAMS,
+              message: `invalid argument ${index}: required field ${field}`,
+            }
+          }
+        }
+
+        const validate = (field: any, validator: Function) => {
+          if (field === undefined) return
+          const v = validator([field], 0)
+          if (v !== undefined) return v
+        }
+
+        // validate pubkey
+        for (const field of [wt.pubkey]) {
+          const v = validate(field, this.bytes48)
+          if (v !== undefined) return v
+        }
+
+        // validate withdrawalCredentials
+        for (const field of [wt.withdrawalCredentials]) {
+          const v = validate(field, this.bytes32)
+          if (v !== undefined) return v
+        }
+
+        // validate amount, index
+        for (const field of [wt.amount, wt.index]) {
+          const v = validate(field, this.bytes8)
```

Comment:
Never mind, https://github.com/ethereum/execution-apis/blob/main/src/engine/prague.md#ExecutionPayloadV4 mentions is as `64 bits`. Which I honestly think is strange since the EIP itself clearly states its an `uint64` and not `bytes8`.